src/vs/base/browser/ui/list/listView.ts

```diff
@@ -447,9 +447,15 @@ export class ListView<T> implements ISpliceable<T>, IDisposable {
 		this.items[index].size = size;
 
 		this.render(lastRenderRange, Math.max(0, this.lastRenderTop + heightDiff), this.lastRenderHeight, undefined, undefined, true);
-		this.setScrollTop(this.lastRenderTop);
 
-		this.eventuallyUpdateScrollDimensions();
+		if (heightDiff > 0) {
+			// update scroll dimension first to otherwise the new scroll top might be invalid
+			this.eventuallyUpdateScrollDimensions();
```

Comment:
It feels weird to me that the order matters here, instead if we have these two properties that depend on each other, they should be set simultaneously and it should be managed in a central spot.